pandas/core/nanops.py

```diff
@@ -812,15 +812,21 @@ def unique1d(values):
     return uniques
 
 
-def _checked_add_with_arr(arr, b):
+def _checked_add_with_arr(arr, b, arr_nans=None, b_nans=None):
```

Comment:
btw, this function is differnt than most in ``nanops.py``. we normally actually just call ``isnull`` directly on the arrays, so we *don't* first send in the ``.view('i8')`` for datetimelikes (this is done in the function). I think you should make this more consistent e.g. ideally this signature would actually be: ``_check_add_with_arr(a, b, skipna=True)``